sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -437,3 +437,278 @@ mod test_kubernetes_bootstrap_token {
         }
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// EvictionHardKey represents a string that contains a valid Kubernetes eviction hard signal.
+/// Available eviction hard signals: https://kubernetes.io/docs/tasks/administer-cluster/out-of-resource/
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct EvictionHardKey {
+    inner: String,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Deserialize)]
+#[serde(rename_all = "lowercase")]
+enum EvitionSignal {
+    #[serde(rename = "memory.available")]
+    MemoryAvailable,
+    #[serde(rename = "nodefs.available")]
+    NodefsAvailable,
+    #[serde(rename = "nodefs.inodesFree")]
+    NodefsInodesFree,
+    #[serde(rename = "imagefs.available")]
+    ImagefsAvailable,
+    #[serde(rename = "imagefs.inodesFree")]
+    ImagefsInodesFree,
+    #[serde(rename = "pid.available")]
+    PidAvailable,
+}
+
+impl TryFrom<&str> for EvictionHardKey {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        serde_plain::from_str::<EvitionSignal>(&input).context(error::InvalidPlainValue {
+            field: "Eviction Hard key",
+        })?;
+        Ok(EvictionHardKey {
+            inner: input.to_string(),
+        })
+    }
+}
+string_impls_for!(EvictionHardKey, "EvictionHardKey");
+
+#[cfg(test)]
+mod test_kubernetes_eviction_hard_key {
+    use super::EvictionHardKey;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn good_eviction_hard_key() {
+        for ok in &[
+            "memory.available",
+            "nodefs.available",
+            "nodefs.inodesFree",
+            "imagefs.available",
+            "imagefs.inodesFree",
+            "pid.available",
+            ] {
+            EvictionHardKey::try_from(*ok).unwrap();
+        }
+    }
+
+    #[test]
+    fn bad_eviction_hard_key() {
+        for err in &["", "storage.available", ".bad", "bad.", &"a".repeat(64)] {
+            EvictionHardKey::try_from(*err).unwrap_err();
+        }
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// EvictionHardValue represents a string that contains a valid Kubernetes eviction threshold quantity
+/// https://kubernetes.io/docs/tasks/administer-cluster/out-of-resource/
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct EvictionHardValue {
+    inner: String,
+}
+
+impl TryFrom<&str> for EvictionHardValue {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+
+        ensure!(
+            input.ends_with("Gi") || input.ends_with("Mi") || input.ends_with("%"),
```

Comment:
this not exactly true. According to [doc](https://kubernetes.io/docs/tasks/administer-cluster/out-of-resource/#eviction-thresholds). "**quantity is the eviction threshold quantity, These tokens must match the quantity representation used by Kubernetes**". There are many other format supported for memory represendation itslef, if you look [here](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-memory). We should check Kubernetes documentation for each quantity type and its supported format. 

I feel this way of parsing would not work, because different types of `EvictionSignal` have their own supported format. You might have to parse based on `EvictionHardKey`. Not sure how we can do that. 

Something that comes to my mind with limited rust knowledge that I have:
```
struct MemoryValue {
    inner: String,
}

struct SomeOtherSignalValue {
    inner: String,
}
enum EvitionSignal {
    MemoryAvailable(MemoryValue),
    PidAvailable(u64),  // Assuming pid only takes integer
    SomeOtherSignalType(SomeOtherSignalValue)
}
```
Also check this for other settings, on first look it looks like above comments is also valid for `kube-reserved` and `system-reserved` settings.

I also feel we should declare type based on Quantiy format instead of setting. So we can use same Type lets say `MemoryValue` for different type of setting parsing.